test_evm_eth_compliance/src/statetest/models/mod.rs

```diff
@@ -0,0 +1,150 @@
+mod deserializer;
+mod spec;
+
+use bytes::Bytes;
+use serde::Deserialize;
+use std::collections::{BTreeMap, HashMap};
+
+use crate::common::{B160, B256, U256};
+
+pub use self::spec::SpecName;
+use deserializer::*;
+
+#[derive(Debug, PartialEq, Eq, Deserialize)]
+pub struct TestSuit(pub BTreeMap<String, TestUnit>);
+
+#[derive(Debug, PartialEq, Eq, Deserialize)]
+pub struct TestUnit {
+    pub env: Env,
+    pub pre: BTreeMap<B160, AccountInfo>,
+    pub post: BTreeMap<SpecName, Vec<Test>>,
+    pub transaction: TransactionParts,
+}
+
+/// State test indexed state result deserialization.
+#[derive(Debug, PartialEq, Eq, Deserialize)]
+pub struct Test {
+    /// Post state hash
+    pub hash: B256,
+    /// Indexes
+    pub indexes: TxPartIndices,
+    // logs
+    pub logs: B256,
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_opt_str_as_bytes")]
+    pub txbytes: Option<Bytes>,
+}
+
+#[derive(Debug, PartialEq, Eq, Deserialize)]
+pub struct TxPartIndices {
+    pub data: usize,
+    pub gas: usize,
+    pub value: usize,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct AccountInfo {
+    pub balance: U256,
+    #[serde(deserialize_with = "deserialize_str_as_bytes")]
+    pub code: Bytes,
+    #[serde(deserialize_with = "deserialize_str_as_u64")]
+    pub nonce: u64,
+    pub storage: HashMap<U256, U256>,
+}
+
+#[derive(Debug, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Env {
```

Comment:
How do you expect to plug in all these environmental inputs? It wouldn't be too crazy to have a fixed Env object in the EVM runtime that can be replaced only when a feature (e.g. `conformance_tests`) is enabled?